src/OnlineSales/Controllers/BaseController.cs

```diff
@@ -153,12 +152,12 @@ protected async Task<T> FindOrThrowNotFound(int id)
         private IQueryProvider<T> BuildQueryProvider(int maxLimitSize)
         {
             var queryCommands = this.Request.QueryString.HasValue ? HttpUtility.UrlDecode(this.Request.QueryString.ToString()).Substring(1).Split('&').ToArray() : new string[0];
-            var parseData = new QueryParseData<T>(queryCommands, maxLimitSize);
+            var parseData = new QueryParseData<T>(queryCommands, maxLimitSize, dbContext);

```

Comment:
IMHO QueryParser shouldn't be linked together with dbContext. His only task to parse query string into query model and validate syntax.
Everything related to figuring out whether query is valid relative to any kind of query provider should be responsibility of  query provider

Reply:
On the one hand, you right. On the other hand, QueryParseData (may be its name is little bit incorrect) ctor doesn't just parse a query expression but try to validate all it can, and I pass dbContext instance in QueryParseData constructor not for handling queries, but for validation some parsed data and this procedure of validation is common and mandatory for any kind of query provider that handles this parsed data. So, I would prefer not to change this piece of code.